### PR TITLE
Add Lachowski quorum intersection checker

### PR DESCRIPTION
**Pull request by liamsi**
A python port of what was integrated into stellar-core via
https://github.com/stellar/stellar-core/pull/2127 https://github.com/stellar/stellar-core/pull/2279

ref:  https://arxiv.org/pdf/1902.06493.pdf and https://github.com/fixxxedpoint/quorum_intersection.git

### CODE REVIEW

**Comment by andrenarchy** [comment]
@Liamsi Can you give a status update of this PR? What's still missing?

**Comment by liamsi** [comment]
First, picking the next node (one with the largest in-degree) needs to be implemented (https://github.com/andrenarchy/stellar-observatory/pull/3/files#diff-50c132b2ed88c2ae576ca5c8397354b3R196); also, the main iteration is not (fully) implemented: https://github.com/andrenarchy/stellar-observatory/pull/3/files#diff-50c132b2ed88c2ae576ca5c8397354b3R149

File: stellarobservatory/quorum_test.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def test_is_quorum():
     """Test is_quorum()"""
     quorum_slices_by_public_key = {
         'A': [{'A', 'B'}, {'A', 'C'}, {'A', 'B', 'C'}],
-        'B': [{'A', 'B'}],
+        'B': [{'A', 'B'}, {'A', 'D'}],
```
Comment (liamsi) [inline stellarobservatory/quorum_test.py:52]: Hmm, not sure why I've changed this. Probably, can be reverted:
```suggestion
        'B': [{'A', 'B'}],
```

File: stellarobservatory/quorum_lachowski_test.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+def test_has_quorum_intersection_false():
+    """Test failing has_quorum_intersection()"""
+    # Basically two disjoint quorums {A, B} and {D} (two SCCs)
+    quorum_slices_by_node = {
+        'A': [{'A', 'B'}, {'A', 'C'}, {'A', 'B', 'C'}],
+        'B': [{'A', 'B'}],
+        'C': [{'A', 'B', 'C', 'D'}],
+        'D': [{'D'}]
+    }
+
```
Comment (liamsi) [inline stellarobservatory/quorum_lachowski_test.py:31]: NIt: for the sake of consistency we could remove this line (the only test where there is a new-line between the case and the assert).
```suggestion
```

**Review by liamsi — COMMENTED** [review]
Looking good 👍 and this was fun, too.

**Pull request merged**